widget/editor.go

```diff
@@ -225,6 +237,82 @@ func (e *Editor) processPointer(gtx layout.Context) {
 			}
 		}
 	}
+
+	for _, evt := range e.dragger.Events(gtx.Metric, gtx, gesture.Both) {
+		// fmt.Printf("Drag event: %+v\n", evt)
+
+		switch {
+		// This case basically augments the work done in the above e.clicker
+		// code.
+		case evt.Type == pointer.Press && evt.Source == pointer.Mouse: /*,
+			evt.Type == gesture.TypeClick && evt.Source == pointer.Touch: */
+			if e.dragging {
+				continue
+			}
+
+			e.startDrag = Point{X: e.caret.col, Y: e.caret.line}
+			e.endDrag = e.startDrag
+			e.dragging = true
+
+		case evt.Type == pointer.Drag && evt.Source == pointer.Mouse:
+			if !e.dragging {
+				continue
+			}
+
+			e.blinkStart = gtx.Now
+			e.moveCoord(image.Point{
+				X: int(math.Round(float64(evt.Position.X))),
+				Y: int(math.Round(float64(evt.Position.Y))),
+			})
+			e.endDrag = Point{X: e.caret.col, Y: e.caret.line}
+			e.caret.scroll = true
+
+		case evt.Type == pointer.Release && evt.Source == pointer.Mouse: /*,
+			/* evt.Type == gesture.TypeClick && evt.Source == pointer.Touch: */
+			if !e.dragging {
+				continue
+			}
+
+			e.blinkStart = gtx.Now
+			e.moveCoord(image.Point{
+				X: int(math.Round(float64(evt.Position.X))),
+				Y: int(math.Round(float64(evt.Position.Y))),
+			})
+			e.endDrag = Point{X: e.caret.col, Y: e.caret.line}
+			e.caret.scroll = true
+			e.dragging = false
+
+			// If they dragged back to where they started, abort.
+			if e.startDrag == e.endDrag {
+				break
+			}
+
+			// Start must be <= end.
+			e.startDrag, e.endDrag = sortPoints(e.startDrag, e.endDrag)
+
+			// Grab the selected text.
+			var selection string
+			if e.startDrag.Y == e.endDrag.Y {
+				selection = e.lines[e.startDrag.Y].Layout.Text[e.startDrag.X:e.endDrag.X]
+			} else {
+				var b strings.Builder
+				b.WriteString(e.lines[e.startDrag.Y].Layout.Text[e.startDrag.X:])
+				for i := e.startDrag.Y + 1; i < e.endDrag.Y; i++ {
+					b.WriteString(e.lines[i].Layout.Text)
+				}
+				b.WriteString(e.lines[e.endDrag.Y].Layout.Text[:e.endDrag.X])
+				selection = b.String()
+			}
+
+			if selection != "" {
+				fmt.Printf("Selection: '%s'\n", selection)
```

Comment:
Remove debugging.

Reply:
Whoops, thought I had.  Will do.

Reply:
The Printf is from the first commit in the PR, and is removed in the second commit.